troubleshooting/debugging.md

```diff
@@ -674,6 +679,95 @@ To work around this issue, we suggest using the following option so that system
+
+If you choose to deploy this, for the S3 upload on shutdown to work, you must:
+
+1. Set the following env vars:
+    a. `S3_BUCKET` => an S3 bucket that your task can upload too. 
+    b. `S3_KEY_PREFIX` => this is the key prefix in S3 for the core dump, set it to something useful like the ticket ID or a human readable string. It must be valid for an S3 key.
+2. You then must add the following S3 permissions to your task role so that the AWS CLI can upload to S3.
+
+```
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Resource": [
+                "arn:aws:s3:::YOUR_BUCKET_NAME",
+                "arn:aws:s3:::YOUR_BUCKET_NAME/*"
+            ],
+            "Effect": "Allow",
+            "Action": [
+                "s3:DeleteObject",
+                "s3:GetBucketLocation",
+                "s3:GetObject",
+                "s3:ListBucket",
+                "s3:PutObject"
+            ]
+        }
+    ]
+}
+```
+
+Make sure to edit the `Resource` section with your bucket name. 
+
+
+##### 2. Setup your own repro attempt 
+
+There are two options for reproducing a crash:
+1. [Tutorial: Replicate an ECS FireLens Task Setup Locally](#tutorial-replicate-an-ecs-firelens-task-setup-locally)
+2. [Template: Replicate FireLens case in Fargate](#template-replicate-firelens-case-in-fargate)
+
+Replicating the case in Fargate is recommended, since you can easily scale up the repro to N instances. This makes it much more likely that you can actually reproduce the issue. 
+
+
+##### Template: Replicate FireLens case in Fargate
+
+In [troubleshooting/tutorials/cloud-firelens-crash-repro-template](tutorials/cloud-firelens-crash-repro-template/), you will find a template for setting up a customer repro in Fargate. 
+
+This can help you quickly setup a repro. The template and this guide require thought and consideration to create an ideal replication of the customer's setup.
+
+Perform the following steps:
+1. Copy the template to a new working directory.
+2. Setup the custom logger. Note: this step is optional, and other tools like [firelens-datajet](https://github.com/aws/firelens-datajet) may be used instead. You need some simulated log producer.
+    1. If the customer tails a file, add their log content or simulated log content to the `file1.log`. If they tail more than 1 file, then add a `file2.log` and add an entry in the `logger.sh` for it. If they do not tail a file, remove the file entry in `logger.sh`.
+    2. If the customer has a TCP input, place their log content or simulated log content in `tcp.log`. If there are multiple TCP inputs or no TCP inputs, customize `logger.sh` accordingly.
```

Comment:
TCP could be more explicit that additional TCP listeners will need additional lines in the shell script to account for it and environment variables or hardcoded values in the task definition